department-of-veterans-affairs/vets-website
# Add Oracle Health sandbox APIs

## Are you removing, renaming or moving a folder in this PR?


### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary
This PR allows the support of two of Oracle Health's staging instances for the unified sign-in page. This effectively allows Oracle Health and other engineers to test functionality between interested parties.

## Related issue(s)
[JIRA VI-512](https://jira.devops.va.gov/browse/VI-512)


## Testing done
Unit testing / manual testing

## Screenshots
n/a

## What areas of the site does it impact?
Unified Sign-in page for Oracle Health

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback
How to test:
1. Navigate to the `/sign-in` page 
2. To test Oracle Health staging instance append the following URL:
    - `?application=myvahealth&to=%2Fsession-api%2Frealm%2Ff0fded0d-d00b-4b28-9190-853247fd9f9d%3Fto%3Dhttps%253A%252F%252Fstaging-patientportal.myhealth.va.gov%252F&oauth=false`
3. To test Oracle Health sandbox instance append the following URL:
    - `?application=myvahealth&to=%2Fsession-api%2Frealm%2Ff0fded0d-d00b-4b28-9190-853247fd9f9d%3Fto%3Dhttps%253A%252F%252Fsandbox-patientportal.myhealth.va.gov%252F&oauth=false`
4. Click a Sign in button > then click the back arrow to take you back to the unified sign-in page
5. Open up DevTools > Application (in Chrome) > Look at the session storage 
6. Confirm based on the instances that the URL points to the correct place
